### PR TITLE
Test fixes and more local mode

### PR DESCRIPTION
**Pull request by baitsguy**
Fix an integ test I broke in https://github.com/aryn-ai/sycamore/commit/0a3ad992a471d68658d0bebc9d50b73b599d092f

Add local mode for limit and sort

### CODE REVIEW

File: lib/sycamore/sycamore/tests/integration/transforms/test_rerank.py [hunk 1]
```diff
@@ -13,39 +14,47 @@ def test_rerank_docset():
...
+                {"properties": {"_element_index": 1}, "text_representation": "here is an animal that meows"},
             ],
         },
         {
             "doc_id": 2,
             "elements": [
-                {"id": 7, "text_representation": "this is a cat"},
-                {"id": 1, "text_representation": "here is an animal that moos"},
+                {"id": 7, "properties": {"_element_index": 7}, "text_representation": "this is a cat"},
+                {"id": 1, "properties": {"_element_index": 1}, "text_representation": "here is an animal that moos"},
```
Comment (HenryL27) [inline lib/sycamore/sycamore/tests/integration/transforms/test_rerank.py:24]: ids here?

File: lib/sycamore/sycamore/tests/integration/transforms/test_rerank.py [hunk 2]
```diff
@@ -13,39 +14,47 @@ def test_rerank_docset():
...
             "elements": [
-                {"text_representation": "here is an animal that moos"},
+                {"properties": {"_element_index": 1}, "text_representation": "here is an animal that moos"},
             ],
         },
         {  # handle element with not text
             "doc_id": 4,
             "elements": [
-                {"id": 1},
+                {"id": 1, "properties": {"_element_index": 1}},
```
Comment (HenryL27) [inline lib/sycamore/sycamore/tests/integration/transforms/test_rerank.py:36]: id here?

File: lib/sycamore/sycamore/tests/integration/transforms/test_rerank.py [hunk 3]
```diff
@@ -13,39 +14,47 @@ def test_rerank_docset():
...
+                    "text_representation": "the number of pages in this document are 253",
+                }
             ],
         },
-        {"doc_id": 5, "elements": [{"text_representation": "the number of pages in this document are 253"}]},
         {  # drop because of limit
             "doc_id": 6,
             "elements": [
-                {"id": 1},
+                {"id": 1, "properties": {"_element_index": 1}},
```
Comment (HenryL27) [inline lib/sycamore/sycamore/tests/integration/transforms/test_rerank.py:51]: id here?

**Review by HenryL27 — COMMENTED** [review]
where id versus _element_id?

**Comment by baitsguy** [comment]
> where id versus _element_id?

they're just no-ops here, I had them to indicate that element_index isn't an id

**Review by HenryL27 — APPROVED** [review]

**Pull request merged**